build/common.sh

```diff
@@ -596,70 +653,134 @@ function kube::build::run_build_command() {
 
   # Clean up container from any previous run
   kube::build::destroy_container "${KUBE_BUILD_CONTAINER_NAME}"
-  "${docker_cmd[@]}" "$@"
+  "${docker_cmd[@]}" "${cmd[@]}"
   kube::build::destroy_container "${KUBE_BUILD_CONTAINER_NAME}"
 }
 
-# Test if the output directory is remote (and can only be accessed through
-# docker) or if it is "local" and we can access the output without going through
-# docker.
-function kube::build::is_output_remote() {
-  rm -f "${LOCAL_OUTPUT_SUBPATH}/test_for_remote"
-  kube::build::run_build_command touch "${REMOTE_OUTPUT_BINPATH}/test_for_remote"
+function kube::build::probe_address {
+  # Apple has an ancient version of netcat with custom timeout flags.  This is
+  # the best way I (jbeda) could find to test for that.
```

Comment:
+1 - I see MacOS as MORE impetus to make the docker-build be default.

On Wed, Aug 31, 2016 at 6:20 PM, Morgan Bauer <notifications@github.com>
wrote:

> In build/common.sh
> <https://github.com/kubernetes/kubernetes/pull/30787#discussion_r77101203>
> :
>
> >    kube::build::destroy_container "${KUBE_BUILD_CONTAINER_NAME}"
> >  }
> >
> > -# Test if the output directory is remote (and can only be accessed through
> > -# docker) or if it is "local" and we can access the output without going through
> > -# docker.
> > -function kube::build::is_output_remote() {
> > -  rm -f "${LOCAL_OUTPUT_SUBPATH}/test_for_remote"
> > -  kube::build::run_build_command touch "${REMOTE_OUTPUT_BINPATH}/test_for_remote"
> > +function kube::build::probe_address {
> > +  # Apple has an ancient version of netcat with custom timeout flags.  This is
> > +  # the best way I (jbeda) could find to test for that.
>
> No actionable comment here.
>
> I am also disappointed in the old tools. If we can't force people to brew
> install, then another reason for moving into a container is to get rid of
> any kind of platform specific scripting.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/30787/files/487945995c84c43caa6f82e902bf277adf89dad2..166d23a43aa2b0f825bc6feec938684b1f43a323#r77101203>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFVgVKFpENUCcO-O8a5FYFVH0ADCcJlkks5qlih7gaJpZM4Jmt7e>
> .
>